File: backend/src/main/java/com/awais/hr/module/visitor/service/VisitorService.java
package com.awais.hr.module.visitor.service;

import java.util.List;
import java.util.Map;

public interface VisitorService {
    List<Map<String, Object>> getVisitors();
    Map<String, Object> registerVisitor(Map<String, Object> body);
    Map<String, Object> checkInVisitor(String visitorId);
    Map<String, Object> checkOutVisitor(String visitorId);
    Map<String, Object> updateVisitorStatus(String visitorId, String status);
}

File: frontend/src/services/visitorService.js
import apiClient from './api';

export const getVisitors = () => apiClient.get('/suite/visitors');
export const registerVisitor = (dto) => apiClient.post('/suite/visitors', dto);
export const checkInVisitor = (id) => apiClient.post(`/suite/visitors/${id}/check-in`, {});
export const checkOutVisitor = (id) => apiClient.post(`/suite/visitors/${id}/check-out`, {});
export const updateVisitorStatus = (id, status) => apiClient.post(`/suite/visitors/${id}/status`, { status });
